### ISSUE TITLE
Overriding original destructor

### ISSUE THREAD

**Issue by lislon** [mockery/mockery#215]
Hello, 

Say I have a class:

```
class Foo
{
   function __construct() { /* open file */ }
   function __destruct()  { /* close file */ } 
}
```

When i mock this class `$foo = Mockery::mock('Foo');` it override original constructor, but not override destructor. So i had errors in the destructor. 

I think Mockery should override destructor & constructor together. I.e. when i use

`$foo = Mockery::mock('Foo', array());` - it uses original class constructor & destructor
`$foo = Mockery::mock('Foo');` - no original constructor & destructor called

**Comment by pscheit** [comment]
is there a workaround for this?

**Comment by davedevelopment** [comment]
@pscheit there was some work done to try and mitigate it in 76f1a56c2, will that work for you?

**Comment by robertbasic** [comment]
This was implemented by #479 

If this is something you still have issue with, please open a new issue with a small, reproducible test case.

Thank you!
